lib/pbench/server/api/resources/users_api.py

```diff
@@ -11,6 +11,57 @@
 from pbench.server.api.auth import Auth
 
 
+class PasswordRecovery(Resource):
+    def __init__(self, logger, auth, mail):
+        self.logger = logger
+        self.auth = auth
+        self.mail = mail
+
+    def post(self):
+        """
+        Post request for initiating password recovery.
+        This requires a JSON data with required user fields
+        {
+            "email": "user@domain.com"
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+
+        :return:
+            Success: 201 with password reset link sent to email
+            Failure: <status_Code>,
+                    response_object = {
+                        "message": "failure message"
+                    }
+        """
+        # get the post data
+        user_data = request.get_json()
+        if not user_data:
+            self.logger.warning("Invalid json object: {}", request.url)
+            abort(HTTPStatus.BAD_REQUEST, message="Invalid json object in request")
+
+        email = user_data.get("email")
+        if not email:
+            self.logger.warning("Missing email field")
+            abort(HTTPStatus.BAD_REQUEST, message="Missing email field")
+
+        # Check if email exists in the Db, else abort
+        try:
+            user = User.query(email=user_data.get("email"))
+        except Exception:
+            self.logger.exception("Exception while querying email")
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+
+        if not user:
+            self.logger.exception("No such user exists with the given email ID")
+            abort(HTTPStatus.BAD_REQUEST, message="No such user exists with the given email ID")
```

Comment:
I like reporting when the email address is unrecognized -- I think it produces a superior user experience -- but I also think it constitutes an information leak, so we probably shouldn't do that.  That is, if we can validate the email address, we'll send the message; otherwise we won't; and, either way, we should return success.  🙁

Reply:
Good point! But in dashboard's perspective if the user has a typo in his email address and he gets a success, now he is waiting for the email and no such email arrives. I think a generic message such as "Invalid email" can be useful here